### ISSUE TITLE
kicking off a $destroy from mouseclick may cause vue devtools to block main thread waiting on stringify() to complete

### ISSUE THREAD

**Issue by rDr4g0n** [vuejs/vue-devtools#464]
I have a dynamically mounted component which is destroyed when the user clicks a cancel button. It appears that in some cases where vue devtools does some stringify / encode that recurses and takes a very long time, blocking the main thread until it is completed. This is accompanied by an error message in the console along the lines of: `app.js:6120 [Vue warn]: Property or method "toJSON" is not defined on the instance but referenced during render. Make sure to declare reactive data properties in the data option. (found in <Root>)`.  

The console error and deep encode code do not occur when the vue devtools tab has not been visited after reloading chrome devtools (I assume it hasnt hooked into vue at that point).  

In my case this occurs when using an elementui select component within my manually mounted component. Destroying my manually mounted component in turn destroys the children. Each el-select (and el-option presumably) that is destroyed adds ~125ms to block the main thread.

Here's part of the flame graph. Those encodes go waaaaayyy down and there are much more to the right. You can see the main thread being blocked for close to 5.5s here. Green is vue.esm, blue is backend.js.

![encode](https://user-images.githubusercontent.com/1927558/33442331-86e78524-d5ba-11e7-9653-fdbe638bb538.png)

I can easily work around the blocking by putting the destroy function call in a set timeout, but I am also a little annoyed by the "toJSON" warnings, so it'd be good to fix them both properly.  

Please let me know if you need more detail. Thanks!

**Comment by rDr4g0n** [comment]
In issue #460 the same problem is encountered when using vuex to hold onto a reference to a component. I am doing the same with this dynamically mounted component. I suspect this issue has to do with recursively encoding a bunch of child components in order to enable vuex time travel?

**Comment by posva** [comment]
Duplicate of #460 
Let's track it there 🙂

**Comment by rDr4g0n** [comment]
@posva sure thing. should I comment on #460 with my findings?
